Desktop_Interface/functiongencontrol.cpp

```diff
@@ -119,67 +126,68 @@ void functionGenControl::waveformName(QString newName, int channelID)
 
     sscanf(divisibilityString, "%d", &channel.divisibility);
 
-    qDebug() << "Length = " << channel.length;
+    qDebug() << "Length = " << length;
     qDebug() << "Divisibility = " << channel.divisibility;
 
-	channel.samples.resize(channel.length);
+	channel.samples.resize(length);
 
-    char *dataString = (char *) malloc(channel.length*5+1);
-    fgets(dataString, channel.length*5+1, fptr);
+    char *dataString = (char *) malloc(length*5+1);
+    fgets(dataString, length*5+1, fptr);
 
     int dummy;
     char *dataStringCurrent = dataString;
-    for (int i = 0; i < channel.length; i++)
+    for (int i = 0; i < length; i++)
 	{
         sscanf(dataStringCurrent, "%d", &dummy);
         dataStringCurrent += strcspn(dataStringCurrent, "\t") + 1;
-        channel.samples[i] = uint8_t(dummy);
+        channel.samples[i] = static_cast<uint8_t>(dummy);
     }
 
     free(dataString);
     fclose(fptr);
 #endif
 
-	double newMaxFreq = DAC_SPS / (channel.length >> (channel.divisibility - 1));
-	double newMinFreq = double(CLOCK_FREQ) / 1024.0 / 65535.0 / channel.length;
+	double newMaxFreq = DAC_SPS / (length >> (channel.divisibility - 1));
+	double newMinFreq = double(CLOCK_FREQ) / 1024.0 / 65535.0 / length;
```

Comment:
Hey man.  Once again, this is my original bad, but it looks like length is not cast to double here.  Float / int = int.